internal/bus/chainsubscriber.go

```diff
@@ -555,6 +589,10 @@ func v1ContractUpdate(fce types.FileContractElement, rev *types.FileContractElem
 		curr.revisionNumber = rev.FileContract.RevisionNumber
 		curr.fileSize = rev.FileContract.Filesize
 	}
+	if curr.revisionNumber == math.MaxUint64 && curr.fileSize == 0 {
```

Comment:
we should probably do a pass of `renterd` and check all spots where we compare to `math.MaxUint64`

Reply:
I did. Actually not that many places. Mostly in rhp2 and rhp3 to return an error. In the contractor we used it to avoid broadcasting a revision for a renewed contract but I changed that to use `RenewedTo` instead.